src/classlibnative/bcltype/system.cpp

```diff
@@ -73,9 +73,11 @@ FCIMPLEND;
 
 
 
-#ifndef FEATURE_CORECLR
 INT64 QCALLTYPE SystemNative::GetWorkingSet()
 {
+#if PLATFORM_UNIX
+    return 0; // TODO: Implement in PAL
```

Comment:
There's potentially another option.  As you know I'd been interested previously in just moving Environment to live in corefx, so that its implementation could be shared across runtimes, so that we could make better use of other implementations in corefx, etc.  The reason for not doing so is some of the implementations in Environment depend on runtime functionality.  What if we were to introduce a "private" EnvironmentCore class, that's exposed publically from System.Private.Corelib and included in the model file, but not available in any contract.  That class would contain only the runtime-related "primitives" necessary to enable everything else.  We'd put the same APIs in corert with whatever implementation needed to live there.  Then Environment would live in corefx, with the majority of the implementation in corefx, calling into EnvironmentCore where needed.  Seems like EnvironmentCore would need to handle:
- Exit and ExitCode
- FailFast
- CommandLine and GetCommandLineArgs
- Get/SetEnvironmentVariable* (due to the caching done by the runtime)
- HasShutdownStarted
- CurrentManagedThreadId
- StackTrace
and the implementation in corefx could implement the rest, plus the actual public APIs exposed from Environment for the above.  It's not perfect, but maybe it would allow for more sharing across runtimes and better leveraging what we already have in corefx.

> The implementation for this exists in corefx already for Process.WorkingSet

I implemented that in corefx and actually forgot it was there ;)  For now I could use the same reflection hack to get at it, if we choose to stay this course.

> If the implementation of  Environment  class lived in corefx repo instead, what would the implementation look like - do we want to make it the same, ignoring the repo boundaries?

The simplest way to reuse it would be reflection.  The implementation right now is split between two builds of System.Diagnostics.Process.dll, one for Linux that uses procfs and one for OSX that uses proc_pidinfo.